tests/integration/test_integration_all_rust.py

```diff
@@ -590,983 +583,1003 @@ def teardown_module():
-        """Test sentry output for autoconnect."""
-        if os.getenv("SKIP_SENTRY"):
-            SkipTest("Skipping sentry test")
-            return
-        # Ensure bad data doesn't throw errors
-        client = AsyncPushTestClient(self._ws_url)
-        await client.connect()
-        await client.hello()
-        await client.send_bad_data()
+
+@pytest.fixture(name="clear_sentry_queue", autouse=True, scope="function")
+def fixture_clear_sentry_queue():
+    """Clear any values present in Sentry queue.
+    Scoped to run automatically after each test function.
+    """
+    yield
+    while not MOCK_SENTRY_QUEUE.empty():
+        MOCK_SENTRY_QUEUE.get_nowait()
+
+
+@pytest.fixture(name="ws_config", scope="session")
+def fixture_ws_config():
+    """Return collection of configuration values."""
+    return {"connection_port": 9150}
+
+
+@pytest.fixture(name="test_client", scope="function")
+async def fixture_test_client(ws_url):
+    """Return a push test client for use within tests that can have
+    a custom websocket url passed in to initialize.
+    """
+    client = AsyncPushTestClient(ws_url)
+    yield client
+    if client.ws:
+        log.debug(f"🐍#### test client at {client.url} disconnected.")
         await client.disconnect()
 
-        # LogCheck does throw an error every time
-        async with httpx.AsyncClient() as httpx_client:
-            await httpx_client.get(f"http://127.0.0.1:{CONNECTION_PORT}/v1/err/crit", timeout=30)
-
-        event1 = MOCK_SENTRY_QUEUE.get(timeout=5)
-        # NOTE: this timeout increased to 5 seconds as was yielding
-        # errors in CI due to potential race condition and lingering
-        # artifact in the MOCK_SENTRY_QUEUE. It could cause test_sentry_output_autoendpoint
-        # to have two LogCheck entires and test_no_sentry_output would have data
-        # when it should not.
-
-        # new autoconnect emits 2 events
-        try:
-            MOCK_SENTRY_QUEUE.get(timeout=5)
-        except Empty:
-            pass
-        assert event1["exception"]["values"][0]["value"] == "LogCheck"
-
-    @pytest.mark.asyncio
-    @max_logs(endpoint=1)
-    async def test_sentry_output_autoendpoint(self):
-        """Test sentry output for autoendpoint."""
-        if os.getenv("SKIP_SENTRY"):
-            SkipTest("Skipping sentry test")
-            return
-        client = await self.quick_register()
-        endpoint = self.host_endpoint(client)
-        await self.shut_down(client)
+
+@pytest.fixture(name="test_client_broadcast", scope="function")
+async def fixture_test_client_broadcast(broadcast_ws_url):
+    """Return a push test client for use within tests that can have
+    a custom websocket url passed in to initialize.
+    """
+    client = AsyncPushTestClient(broadcast_ws_url)
+    yield client
+    await client.disconnect()
+
+
+@pytest.fixture(name="registered_test_client", scope="function")
+async def fixture_registered_test_client(ws_config):
```

Comment:
nit:

```suggestion
async def fixture_registered_test_client(ws_config) -> AsyncPushTestClient:
```

Reply:
This actually returns an `AsyncGenerator` because of the use of `yield`, which is not very helpful from a typing context, which is why I left it out, but no harm in adding it if you like though.